### ISSUE TITLE
V-USB implementation is problematic with purejavahidapi

### ISSUE THREAD

**Issue by vogt31337** [nyholku/purejavahidapi#25]
Hi,

I just tried one of these USB Hid relays (http://vusb.wikidot.com/project:driver-less-usb-relays-hid-interface) and purejavahidapi crashes on:
´´´java
PureJavaHidApi.openDevice(devInfo.getPath());
´´´
It crashes because the implementation on the ATTiny45 returns an empty InputReportByteLength.
Small workaround for HidDevice.java (line 106):
```java
m_InputReportLength = (caps.InputReportByteLength == 0) ? 1 : caps.InputReportByteLength;
```

this will set a length of one if this field is empty. I'm not an expert in USB and I don't know if this will mess things up, or if the variable m_InputReportMemory should just be null. But this hack seems to work with this USB device.

**Comment by nyholku** [comment]
Hi, thanks for the feedback. I thought we had this covered but I will check the code maybe it was not fixed in all backends, which platform are you on?

**Comment by vogt31337** [comment]
```code
java -version
java version "1.8.0_60"
Java(TM) SE Runtime Environment (build 1.8.0_60-b27)
Java HotSpot(TM) 64-Bit Server VM (build 25.60-b23, mixed mode)
```
I'm running on windows 7 x64. It crashes because it tries to new Memory(m_InputReportLength); which fails if it's zero. But I think the Issue will also happen on Linux.

**Comment by nyholku** [comment]
This should be now fixed with the latest release 0.0.2
